core-api/src/main/java/com/optimizely/ab/config/audience/match/MatchTypeError.java

```diff
@@ -0,0 +1,35 @@
+/**
+ *
+ *    Copyright 2019, Optimizely and contributors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.optimizely.ab.config.audience.match;
+
+public enum MatchTypeError {
+    UNKNOWN_MATCH_TYPE("uses an unknown match type. You may need to upgrade to a newer release of the Optimizely SDK"),
+    INAPPLICABLE_CONDITION_VALUE("has an unknown condition value type. You may need to upgrade to a newer release of the Optimizely SDK");
```

Comment:
Thanks for explaining what `INAPPLICABLE_CONDITION_VALUE` is for! Can we change the message a little bit to clarify? I think this would be sufficient:
```
"has an unexpected value type. You may need to upgrade to a newer release of the Optimizely SDK"
```
After thinking about it for a while, I guess it's okay to suggest that the customer upgrade their SDK here. It could help if we eventually support non-primitive values for `exact` match, or lexicographic `gt`/`lt` semantics for strings values, etc. 🤷